### ISSUE TITLE
[Bug report]ValueError: Dimension, `np.size(x)==4`, of input `x` does not match the problem dimension `number_of_variables==2`.

### ISSUE THREAD

**Issue by icygreen-xy** [numbbo/coco#2198]
**Describe the bug**
**Bug-1:** ValueError: Dimension, `np.size(x)==4`, of input `x` does not match the problem dimension `number_of_variables==2`.

I am not sure which one belongs to cocoex module: `np.size(x)==4` or `number_of_variables==2`

because I can not find these two parameters in my new adding solver. Thanks a lot! 

![image](https://user-images.githubusercontent.com/110183414/231184437-17c87046-7f9e-48f7-be8f-1e0dadff4d26.png)

**Bug-2:** when I set  `suite_year_option = "year:2023"` , my `kernel dead always. When I set  `suite_year_option = ""`, this part can run successfully. So should I not set this for 2023 or use cloud computing?

`![image](https://user-images.githubusercontent.com/110183414/231184857-03a02275-4d38-4c34-945d-3969492748b3.png)

**Your System (please complete the following information when it seems relevant):**
 - Operating system: macOS 13.2.1 (22D68)
 - COCO version: code-experiments/build/python/example_experiment2.py
 - Browser (if relevant):  safari

**Comment by icygreen-xy** [comment]
Hi professors,
Can I submit an abstract tomorrow and then submit a full text next week? - I can not deal this bug, and I might need some time on dealing with it.  Thanks so much! 


I change the parameter `dimension `tonight, there are some new value errors to be produced.

```
dimension = 2,  np.size(x) = 4, number_of _variables ==2
dimension =3, np.size(x)= 9, number_of_ variables ==3
dimension =5,np.size(x)= 25, number_of_ variables ==5
dimension =10,np.size(x)= 100, number_of_ variables ==10
```

**Comment by brockho** [comment]
Hello,

It looks like you are calling the evaluate function of COCO with a wrongly-dimensioned argument. It takes a single vector of `problem.dimension` many variables and outputs the corresponding function value (or function value vector in the case of multiobjective optimization). I would not count on the fact that it can evaluate more than a single solution at a time.

**Comment by icygreen-xy** [comment]
This is my setting in parameters, and I use `bbob `suite:

I am not sure what other information can I need to provide. 

So is it about setting too many dimensions: 2,3,5,10 or 40.  When I only set `dimensions = 2`, I still get this` Value Error`.

![image](https://user-images.githubusercontent.com/110183414/231716861-74afd95c-97f8-4f04-8e4f-7cde162612da.png)

![image](https://user-images.githubusercontent.com/110183414/231716935-d0d08e93-418b-4fe6-8f19-5504bf16269c.png)

![image](https://user-images.githubusercontent.com/110183414/231716134-ef2d9e22-6174-4fdd-870b-2ab3be2092a4.png)

![image](https://user-images.githubusercontent.com/110183414/231716322-cb1e6536-1839-4265-a3b1-9dff5ed62ef4.png)

**Comment by ttusar** [comment]
I realize this is very annoying for the user, but it should say `dimensions: 2`, not `dimensions:2` (the space after the colon is missing).

**Comment by icygreen-xy** [comment]
Many thanks, Prof.

I still get results of the error:

![image](https://user-images.githubusercontent.com/110183414/231721650-1e252b43-f506-4528-bc35-af31492ee210.png)


I try to test the last paragraph and get this result:

![image](https://user-images.githubusercontent.com/110183414/231721849-7ef9f1d2-030a-424f-a622-5d98bd97c6d1.png)


![image](https://user-images.githubusercontent.com/110183414/231721938-6bf50c61-bd56-4f96-a115-9d8c755e575e.png)

**Comment by ttusar** [comment]
From the code snippets you provided it is not clear where you set the problem dimension in your algorithm. It seems your algorithm is producing a 4-D solution for the 2-D problem.

**Comment by icygreen-xy** [comment]
I am looking for line 71:`---> 71 self.fitness[i] = self.objf(self.Positions[i, :])` and have a check on my setting of `self.objf` and `self.Positions`.

I doubt the problem might happen in`self.objf`

My codes about `self.objf = objf` and `line 71`
```
self.Positions = np.asarray(
            [x * (self.ub - self.lb) + self.lb for x in np.random.uniform(0, 1, (self.SearchAgents_no, self.dim))]
        )
```
`self.fitness = np.full(self.SearchAgents_no, np.inf)`


```
 # Check boundaries
 self.Positions[i, :] = np.clip(self.Positions[i, :], self.lb, self.ub)
 # fitness of locations
 self.fitness[i] = self.objf(self.Positions[i, :])
```

**Comment by brockho** [comment]
```python
In [1]: import cocoex
In [2]: mysuite = cocoex.Suite("bbob", "", "")
In [3]: ff = mysuite.get_problem_by_function_dimension_instance(10, 2, 4)  # Ellipsoid function in 2-D, instance 4
In [4]: ff([0,0])
Out[4]: 7988460.874516963
```

**Comment by brockho** [comment]
Regarding your question about a deadline extension: we will keep the submission system for the BBOB-2023 workshop open until Monday, April 17, 2023 anywhere on earth. After this time, the submission system will not allow you to submit or change your paper anymore. We will be happy if you can solve your issues until then.

**Comment by icygreen-xy** [comment]
Many thanks, Profs! I am looking for dealing with them
:D

**Comment by icygreen-xy** [comment]
> ```python
> In [1]: import cocoex
> In [2]: mysuite = cocoex.Suite("bbob", "", "")
> In [3]: ff = mysuite.get_problem_by_function_dimension_instance(10, 2, 4)  # Ellipsoid function in 2-D, instance 4
> In [4]: ff([0,0])
> Out[4]: 7988460.874516963
> ```

Hi Prof,

I used the code you provided and got the results you showed. Thank you! However, I would like to learn more about the parameter called `'problem'` in any fmin (solver), which has the same functions and settings as `'get_problem_by_function_dimension_instance()'.`

![image](https://user-images.githubusercontent.com/110183414/232265971-fb63babd-c91b-46c8-b860-662c9930797d.png)



**Screenshot1:** **Get the same result** 

In [1]: import cocoex as ex
In [2]: mysuite = =ex.Suite("bbob", "", "")
In [3]: ff = mysuite.get_problem_by_function_dimension_instance(10, 2, 4)  # Ellipsoid function in 2-D, instance 4
In [4]: ff([0,0])
Out[4]: 7988460.874516963

![image](https://user-images.githubusercontent.com/110183414/232272103-f879da8b-9b5e-4ab9-a556-5ec0ddb5ea77.png)

I know each parameter here represents the function index, dim index and instance index. ff([ paras ]) here means each position in coordinates with different dimensions. so 2 dimensions mean 2 paras in ff([a,b]), then I have **one question** inferred by referring to **screenshot 2**: I called `self.objf = self.problem `( assume the `problem` has the same function as `get_problem_by_function_dimension_instance()`, or does it has more functions?)

**Screenshot2:**  **Get the same error in the result: they are similar in form** 

![image](https://user-images.githubusercontent.com/110183414/232272053-9fc8b879-6ede-469d-851e-87932772d9aa.png)

`para: get_problem_by_function_dimension_instance()` -- 3 dimensions, however, in our previous setting in objective function -  get_problem_by_function_dimension_instance() - 5 dimensions.

![image](https://user-images.githubusercontent.com/110183414/232272065-222287c4-5e12-4f15-b728-d04b6a022ca8.png)

I doubt that there is an input of 4 parameters in `para: problem` -- 4 dimensions, however, in my previous setting in objective function - problem - 2 dimensions. 

So the problem here is that `self.Positions[i, :]` was an array with 4 paras in each list, an array includes many lists.

**Comment by icygreen-xy** [comment]
I am so clear about this problem: 

**If `Problem` ( has the same function as `get_problem_by_function_dimension_instance`, actually this problem can be solved.** 

I tested my code about `71 self.fitness[i] = self.objf(self.Positions[i, :])`
`self.Positions[i, :]` and `self.fitness[i]` see what anything can be produced? 

```python
import numpy as np

ub = 5
lb = -5
SearchAgents_no = 10
dim = 2

Positions = np.asarray([x * (ub - lb) + lb for x in np.random.uniform(0, 1, (SearchAgents_no, dim))])

fitness = np.full(SearchAgents_no, np.inf)




```

**Comment by nikohansen** [comment]
The method `get_problem_by_function_dimension_instance` returns what we call a `problem` which is a function instance which can _only be evaluated in the given dimension_. Iterating over a `suite` iterates over all these problems. Not sure if this addresses your question...

**Comment by icygreen-xy** [comment]
I noticed in `elif fmin in (cocoex.solvers.random_search, random_search):`
so fun = problem here:
![image](https://user-images.githubusercontent.com/110183414/232312410-66f4cc43-1b92-481c-a6d6-e27beb7d6ee7.png)

fun. number_of_constraints = `problem. number_of_constraints` ( I saw it in this link: [
](https://numbbo.github.io/coco-doc/apidocs/cocoex/cocoex.Problem.html#free), but I did not find other thing about `problem. @property` has the same function as the method `get_problem_by_function_dimension_instance`, s**o can I use 
`get_problem_by_function_dimension_instance` to replace `problem.@property` in my function** - my defined function is that 
Run(ff, problem.lower_bounds, problem.upper_bounds, problem.dimension, NumOfRuns=1, popSize=50,Iter=100)

In [1]: import cocoex as ex
In [2]: mysuite = =ex.Suite("bbob", "", "")
In [3]: ff = mysuite.get_problem_by_function_dimension_instance(10, 2, 4) # Ellipsoid function in 2-D, instance 4
In [4]: ff([0,0])
Out[4]: 7988460.874516963

ff means: a position as a coordinate with different dimensions


![image](https://user-images.githubusercontent.com/110183414/232312550-29487d48-68a0-431f-bed1-6f8273c45cc3.png)

**Comment by icygreen-xy** [comment]
I saw it in this link:[ ](https://numbbo.github.io/coco-doc/apidocs/cocoex/cocoex.Problem.html#free)

**Comment by brockho** [comment]
I am not sure where you are stuck right now. You sent the link to the documentation yourself, which says "The main feature of a problem instance is that it is callable, returning the objective function value when called with a candidate solution as input.

It provides other useful properties and methods like [dimension](https://numbbo.github.io/coco-doc/apidocs/cocoex/cocoex.Problem.html#dimension), [number_of_constraints](https://numbbo.github.io/coco-doc/apidocs/cocoex/cocoex.Problem.html#number_of_constraints), [observe_with](https://numbbo.github.io/coco-doc/apidocs/cocoex/cocoex.Problem.html#observe_with), [initial_solution_proposal](https://numbbo.github.io/coco-doc/apidocs/cocoex/cocoex.Problem.html#initial_solution_proposal)..."

**Comment by brockho** [comment]
To compare the different algorithms, you will have to use the postprocessing module of COCO:
```python
python -m cocopp FOLDER1 FOLDER2 FOLDER3
```
for example. Instead of a folder (containing a single experiment), you can also use any name in the [data archive](https://numbbo.github.io/data-archive/) such as `slsqp-11` or `BFGS-scipy`.

**Comment by icygreen-xy** [comment]
Hi Profs,
My PDF is over 25 MB and the system can not accept, so I update a word is 21.3MB. I am not sure why my PDF is so large.

**Comment by icygreen-xy** [comment]
![image](https://user-images.githubusercontent.com/110183414/232774506-19b1d0d9-e9df-4a0d-975f-eb14abc6aa6e.png)

**Comment by icygreen-xy** [comment]
> I got a successful result 😆 when I changed the lower bound and up bound, and I set up a fixed number there, in my algorithm - there was already an ub and lb array generator by using dimensions. So I can not use `problem.lower_bounds`, `problem.upper_bounds` for more information, I'll update after tonight.

my function is `def MABrun(objectivefunctions, lb, ub, dim, NumOfRuns, popSize,Iterations):` in encapsulation - this is a call of main optimizer herustic multiple-armed bandit. 

`objectivefunction` = `problem`
`lb` = `problem.upper_bounds` -- **I will explain this one by codes**
`ub` = `problem.upper_bounds` -- **I will explain this one by codes**
`dim `= `problem.dimension`

`NumOfRuns `= `self defined parameters∈ N `
`popSize =` `self defined parameters∈ N`
`Iterations `= `self defined parameters∈ N`

**Comment by icygreen-xy** [comment]
I ran cocoex module individually and found this problem:
![image](https://user-images.githubusercontent.com/110183414/233669533-b81cfb5b-5960-41ce-9f6d-83cf8cdc316f.png)

Here it is easy to understand: [-5,5]^n ( n =2,3,4,10,20,40)

So `Positions = np.asarray([x * (ub - lb) + lb for x in np.random.uniform(0, 1, (SearchAgents_no, dim))])` can replace
`problem.upper_bounds` and `problem.upper_bounds`

For example, I use `problem.upper_bounds` and `problem.upper_bounds` , there was a bug of `ValueError: Dimension, ‘ np.size(x) ==4’, of input ‘ x’ does not match the problem dimension ‘number _of_ variables ==2`

It also produce a error, assump to use p`roblem.upper_bounds `and `problem.upper_bounds`, when `n =2`.

`ub = [5,5] lb =[-5.-5]`


![image](https://user-images.githubusercontent.com/110183414/233674601-0c6081d6-caff-476c-95f0-0065f6dad17a.png)

**Comment by icygreen-xy** [comment]
np.random.uniform(0, 1, (SearchAgents_no, dim)):


 
```python
import numpy as np

# Generate a 2x3 array with elements in the range [0, 1).

arr =  np.random.uniform((low=0.0, high=1.0, size=(2, 3))
print(arr)

array([[0.50210157, 0.64499455, 0.34316675],
       [0.08678106, 0.27441421, 0.46105586]])

``` 

so when you input -5 and 5, this  np.random.uniform can generate a arrary for your problem.bound by using dims =2,3,4,5,10,20,40.

[x1,x2]
[x1,x2,x3]
[x1,x2,x3,x4]
.......

`Positions = np.asarray([x * (ub - lb) + lb for x in np.random.uniform(0, 1, (SearchAgents_no, dim))])`



![image](https://user-images.githubusercontent.com/110183414/233679844-3d5ab40b-d510-4644-bf6f-fdfe4d6d65e6.png)

![image](https://user-images.githubusercontent.com/110183414/233679793-1f41784d-0def-410a-9820-3d201d4b609c.png)

![image](https://user-images.githubusercontent.com/110183414/233680044-d865518e-9ab9-40a3-a0c5-778f55d92f2a.png)

I am not sure ' SearchAgents_no' produce the row of array will influence the corresponding `problem` -- I will check tomorrow.

**Comment by icygreen-xy** [comment]
`SearchAgents_no` produce the row of array will influence the corresponding `problem`:
Here I mean  `SearchAgents_no` determines the row : it will also determine the number of arrays of `lower bounds (lb)` and `upper bounds (ub)` corresponding to the `dimension (dim)`.

I am wondering there is problem cause a bug or problem for  #2202, but it looks like incorrect. 

1) **I made an assumption - if this problem is true, then the following bugs will not occur.**  -- This test can be visualized in the following part, but seemingly this problem was tested as wrong, and assumption was wrong.

2) change my codes, not use 
`Positions = np.asarray([x * (ub - lb) + lb for x in np.random.uniform(0, 1, (SearchAgents_no, dim))])` to generate an arrary of lb and up corresponding to dims.  -- Upload tomorrow without using this code 


 - there is an clear explanation by using visulization:

**Comment by brockho** [comment]
It becomes more and more clear to me that the error does not come from our side, so shall we close this issue?

**Comment by icygreen-xy** [comment]
Maybe we can close this issue. Sure.

`1 `I've tested without no problem, and I am tested post-processing, if there is a bug, and I'll reply in #2202  and change  my codes  -- `2.change my codes, not use`

1. I made an assumption - if this problem is true, then the following bugs will not occur. -- This test can be visualized in the following part, but seemingly this problem was tested as wrong, and assumption was wrong.


2.change my codes, not use
Positions = np.asarray([x * (ub - lb) + lb for x in np.random.uniform(0, 1, (SearchAgents_no, dim))]) to generate an arrary of lb and up corresponding to dims. -- Upload tomorrow without using this code
